yt/yt/server/query_tracker/yql_engine.cpp

```diff
@@ -40,13 +40,16 @@ class TYqlSettings
 {
 public:
     std::optional<TString> Stage;
+    std::optional<int> ExecutionMode;
```

Comment:
Can we use enum here?

Reply:
I agree with Grigory. In YT code we tend to use enums wherever possible, even if the underlying system operates with int. An example from CHYT world: https://github.com/ytsaurus/ytsaurus/blob/main/yt/chyt/server/private.h#L127-L138. 

This interface is the last mile before passing values to essentially a third-party code via C ABI, so having ENUM here and upwards from here is perfectly fine. Also, we would not want a strange integer parameter in our own API (e.g. in HTTP API it should be enum, which is serialized/deserialized as a string literal).

Reply:
Changed to enum

Seems that registrar can handle enums well, didn't expect that